internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -204,6 +195,36 @@ func (r *Reconciler) getManagedLabels(labels map[string]string) map[string]strin
 	return managedLabels
 }
 
+// getManagedAnnotations returns a map of Node annotations managed by CAPI.
+// This is not generalized with getManagedLabels because the domains used by labels and annotations are slightly different.
+func (r *Reconciler) getManagedAnnotations(machine *clusterv1.Machine) map[string]string {
+	// Always sync CAPI's bookkeeping annotations
+	managedAnnotations := map[string]string{
+		clusterv1.ClusterNameAnnotation:      machine.Spec.ClusterName,
+		clusterv1.ClusterNamespaceAnnotation: machine.GetNamespace(),
+		clusterv1.MachineAnnotation:          machine.Name,
+	}
+	if owner := metav1.GetControllerOfNoCopy(machine); owner != nil {
+		managedAnnotations[clusterv1.OwnerKindAnnotation] = owner.Kind
+		managedAnnotations[clusterv1.OwnerNameAnnotation] = owner.Name
+	}
+	for key, value := range machine.GetAnnotations() {
+		// Always sync CAPI's default label node domain
+		dnsSubdomainOrName := strings.Split(key, "/")[0]
+		if dnsSubdomainOrName == clusterv1.ManagedNodeAnnotationDomain || strings.HasSuffix(dnsSubdomainOrName, "."+clusterv1.ManagedNodeAnnotationDomain) {
+			managedAnnotations[key] = value
```

Comment:
```suggestion
			managedAnnotations[key] = value
			break
```
Same for the 3 cases in getManagedLabels